hare/broker.go

```diff
@@ -138,7 +103,13 @@ func (b *Broker) HandleMessage(ctx context.Context, _ p2p.Peer, msg []byte) pubs
 
 func (b *Broker) handleMessage(ctx context.Context, msg []byte) error {
 	h := types.CalcMessageHash12(msg, pubsub.HareProtocol)
-	logger := b.WithContext(ctx).WithFields(log.Stringer("latest_layer", b.getLatestLayer()), h)
+	var logger log.Log
+	func() {
+		b.mu.RLock()
+		defer b.mu.RUnlock()
+		logger = b.WithContext(ctx).WithFields(log.Stringer("latest_layer", b.latestLayer), h)
+	}()
```

Comment:
this is def. clever.
but i must over-using anonymous func. makes the code harder to read.